src/librbd/Journal.cc

```diff
@@ -471,6 +624,31 @@ int Journal<I>::get_tag_owner(IoCtx& io_ctx, std::string& image_id,
 }
 
 template <typename I>
+void Journal<I>::is_tag_owner(I *image_ctx, ContextWQ *work_queue,
```

Comment:
@trociny Sorry -- yeah, that was in Venky's PR branch. I definitely agree that it would be nicer to just directly extract the values instead of invoking a heavier weight journal open/close. I just have a feeling like the current proposal involves a lot of duplicated logic since it is essentially the first couple states in the "librbd::Journal" state machine. What if you extracted those out to a new "librbd::journal::OpenJournalerRequest" state machine that could be re-used by your logic and "librbd::Journal::create_journaler"?